src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> list = null;
+        if (type == Bulldozer.class) {
+            list = new ArrayList<>(new BulldozerProducer().get());
+        }
+        if (type == Excavator.class) {
+            list = new ArrayList<>(new ExcavatorProducer().get());
+        }
+        if (type == Truck.class) {
+            list = new ArrayList<>(new TruckProducer().get());
+        }
+
+        return list;
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        machines.set(0, value);
```

Comment:
this method should fill the whole machines list with the given value, not just element on the first index